### PR TITLE
Ft revise heatmap

### PR DESCRIPTION
**Pull request by lilly-muyama**
# <Reshape Heatmap >

**_WHAT DOES THIS PR DO?_**
This PR revises the predicted values so that they are inside the boundaries of the region of interest.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A

**_WHAT IS THE LINK TO THE PR BRANCH_**

**_HOW DO I TEST OUT THIS PR?_**
- Git clone this repo
- cd predict
- env\scripts\activate for Windows or source env/bin/activate for MacOS
- pip install -r requirements.txt
- python gcp_helpers/cloud_function.py
- flask run

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
http://127.0.0.1:5000/api/v1/predict/heatmap

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**

**_ARE THERE ANY RELATED PRs?_**

### CODE REVIEW

**Review by Baalmart — COMMENTED** [review]
Thanks, I am getting this error on installation of requirements...could you assist? @lilly-muyama 
Screenshot 2021-07-14 at 17 04 36

**Comment by danielogen** [comment]
Hi @lilly-muyama, I am also stuck at requirements installation. I was able to handle `GDAL` package but now am getting more errors. Any tip will be very helpful.

Screen Shot 2021-07-14 at 5 51 47 PM

**Comment by lilly-muyama** [comment]
> 
> 
> Thanks, I am getting this error on installation of requirements...could you assist? @lilly-muyama
> Screenshot 2021-07-14 at 17 04 36

@Baalmart I don't understand this error because that is the latest version of pyproj. @danielogen @ja-odur have you encountered anything similar? That said, these geo packages always give issues, and people had problems running the location-registry microservice which had the same packages until we used the docker file. If the issue persists, that might be the way to go.

**Comment by lilly-muyama** [comment]
> 
> 
> Hi @lilly-muyama, I am also stuck at requirements installation. I was able to handle `GDAL` package but now am getting more errors. Any tip will be very helpful.
> 
> Screen Shot 2021-07-14 at 5 51 47 PM

@danielogen can you try downgrading the googleresumable media to 0.5.1

**Comment by danielogen** [comment]
> > Hi @lilly-muyama, I am also stuck at requirements installation. I was able to handle `GDAL` package but now am getting more errors. Any tip will be very helpful.
> > Screen Shot 2021-07-14 at 5 51 47 PM
> 
> @danielogen can you try downgrading the google-resumable-media to 0.5.1

@lilly-muyama, I succeeded running this PR on `macOs` by doing the following:

- Install `gdal` using `brew`: `brew install gdal`. Please not that this takes a long while to complete  :)
- Remove version number from the following packages: `google-resumable-media`, `google-cloud-bigquery` and `google-cloud-storage`.

cc: @Baalmart, @NoahNsimbe, @ja-odur

**Comment by danielogen** [comment]
> > > Hi @lilly-muyama, I am also stuck at requirements installation. I was able to handle `GDAL` package but now am getting more errors. Any tip will be very helpful.
> > > Screen Shot 2021-07-14 at 5 51 47 PM
> > 
> > 
> > @danielogen can you try downgrading the google-resumable-media to 0.5.1
> 
> @lilly-muyama, I succeeded running this PR on `macOs` by doing the following:
> 
> * Install `gdal` using `brew`: `brew install gdal`. Please not that this takes a long while to complete  :)
> * Remove version number from the following packages: `google-resumable-media`, `google-cloud-bigquery` and `google-cloud-storage`.
> 
> I have also edited the PR description to correct the `endpoint` route
> 
> cc: @Baalmart, @NoahNsimbe, @ja-odur

@lilly-muyama , could you share the sample output of the endpoint? or this can only be viewed from the frontend?

File: src/predict/gcp_helpers/cloud_function.py [hunk 1]
```diff
@@ -16,12 +16,17 @@
 from pymongo import MongoClient
```
Comment (danielogen) [inline src/predict/gcp_helpers/cloud_function.py:16]: For easy maintainability, we could move the code to its own microservice. Details in this PR: https://github.com/airqo-platform/AirQo-api/pull/431.

How are we handling `tenancy` and `envs`?

File: src/predict/greater_kampala_metropolitan_area_kiidp_2012/Greater_Kampala_Metropolitan_Area_KIIDP_2012.shp.xml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Baalmart) [inline src/predict/greater_kampala_metropolitan_area_kiidp_2012/Greater_Kampala_Metropolitan_Area_KIIDP_2012.shp.xml:1]: Just curious, could we explore the storage of these kinds of files in GCS or any other appropriate tool?
Same case for the binary files I have seen above in the PR......

File: src/predict/requirements.txt [hunk 3]
```diff
@@ -38,42 +44,55 @@ gunicorn==20.0.4
...
 protobuf==3.12.2
 py==1.9.0
+pyarrow==3.0.0
 pyasn1==0.4.8
 pyasn1-modules==0.2.8
 pycparser==2.20
 pymongo==3.10.1
 pyparsing==2.4.7
-pytest==4.6.11
+pyproj==3.1.0
```
Comment (Baalmart) [inline src/predict/requirements.txt:80]: I guess this is the requirement I failed to install. Could it be the version number? Or could it be something wrong with my DEV environment?

On a side note, the requirements seem to be many, are all of them being utilised? If yes, never mind.... :)

**Review by Baalmart — COMMENTED** [review]
Hi @lilly-muyama , I have just left some more comments.  I will continue reviewing while sharing feedback accordingly. Thanks!

**Comment by lilly-muyama** [comment]
> 
> 
> > > > Hi @lilly-muyama, I am also stuck at requirements installation. I was able to handle `GDAL` package but now am getting more errors. Any tip will be very helpful.
> > > > Screen Shot 2021-07-14 at 5 51 47 PM
> > > 
> > > 
> > > @danielogen can you try downgrading the google-resumable-media to 0.5.1
> > 
> > 
> > @lilly-muyama, I succeeded running this PR on `macOs` by doing the following:
> > 
> > * Install `gdal` using `brew`: `brew install gdal`. Please not that this takes a long while to complete  :)
> > * Remove version number from the following packages: `google-resumable-media`, `google-cloud-bigquery` and `google-cloud-storage`.
> > 
> > I have also edited the PR description to correct the `endpoint` route
> > cc: @Baalmart, @NoahNsimbe, @ja-odur
> 
> @lilly-muyama , could you share the sample output of the endpoint? or this can only be viewed from the frontend?

@danielogen the output is similar to the one before. But while predictions were for 10,000 points before, the endpoint now returns predictions for less than 5000 points i.e. only those within the Kampala region as defined by the shape file

File: src/predict/requirements.txt (hunk 3 above)
Comment (lilly-muyama) [reply src/predict/requirements.txt:80]: There is nothing wrong with the version as it exists but I think a Dockerfile will e the best option to run this once it is its own microservice

**Comment by danielogen** [comment]
Hi @lilly-muyama , is the PR still valid?

**Pull request closed without merging**